File: kotlin/isogram/src/main/kotlin/Isogram.kt
class Isogram {
    companion object {
        fun isIsogram(s: String): Boolean {
            val justChars = s.toLowerCase().replace(Regex("[^a-z]"), "")
            val uniqueChars = justChars.toHashSet()

            return uniqueChars.size == justChars.length
        }
    }
}

File: typescript/isogram/isogram.ts
class Isogram {
    static isIsogram( word: string ): boolean {
        let lowerCaseCharsOnly = word.toLowerCase().match(/[a-z]/g);
        let uniqueChars = new Set(lowerCaseCharsOnly);
        
        return (lowerCaseCharsOnly || "").length === uniqueChars.size;
    }
}

export default Isogram
